measureme/src/profiling_data.rs

```diff
@@ -152,3 +169,234 @@ impl<'a> Iterator for MatchingEventsIterator<'a> {
         None
     }
 }
+
+/// A `ProfilingDataBuilder` allows for programmatically building
+/// `ProfilingData` objects. This is useful for writing tests that expect
+/// `ProfilingData` with predictable events (and especially timestamps) in it.
+///
+/// `ProfilingDataBuilder` provides a convenient interface but its
+/// implementation might not be efficient, which why it should only be used for
+/// writing tests and other things that are not performance sensitive.
+pub struct ProfilingDataBuilder {
+    event_sink: Arc<ByteVecSink>,
```

Comment:
Do we need to use `Arc`s here? Unit tests should be single threaded so we don't need this to work across multiple threads.

Reply:
You're right. I basically copied this from the string table case, but here a plain EventSink should work just fine.

Reply:
I removed the `Arc`.